### PR TITLE
13134: Labeler cleanup, rendering improvements

### PR DESCRIPTION
**Pull request by uckelman**

### CODE REVIEW

File: vassal-app/src/main/java/VASSAL/counters/Labeler.java [hunk 1]
```diff
@@ -262,56 +276,11 @@ public String getLocalizedName() {
...
-// FIXME: This doesn't belong here. Should be in ImageUtils instead?
-  public static void drawLabel(Graphics g, String text, int x, int y, int hAlign, int vAlign, Color fgColor, Color bgColor) {
-    drawLabel(g, text, x, y, new Font("Dialog", Font.PLAIN, 10), hAlign, vAlign, fgColor, bgColor, null);
-  }
-
-// FIXME: This doesn't belong here. Should be in ImageUtils instead?
-  public static void drawLabel(Graphics g, String text, int x, int y, Font f, int hAlign, int vAlign, Color fgColor, Color bgColor, Color borderColor) {
-    g.setFont(f);
-    final int width = g.getFontMetrics().stringWidth(text + "  ");
-    final int height = g.getFontMetrics().getHeight();
```
Comment (BrentEaston) [inline vassal-app/src/main/java/VASSAL/counters/Labeler.java:275]: Shouldn't these be deprecated and defer to LabelUtils.drawLabel()? I use these all over the place in my custom code as a quick & nasty to paste odd bits of info on top of counters.

Comment (BrentEaston) [reply vassal-app/src/main/java/VASSAL/counters/Labeler.java:275]: Ah Sorry, didn't scroll down far enough. Looks good.

File: vassal-app/src/main/java/VASSAL/tools/image/LabelUtils.java [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+    if (borderColor != null) {
+      g.setColor(borderColor);
+      g.drawRect(x0, y0, width, height);
+    }
+
+    g.setColor(fgColor);
+    ((Graphics2D) g).setRenderingHint(RenderingHints.KEY_ANTIALIASING,
+                                      RenderingHints.VALUE_ANTIALIAS_ON);
+    g.drawString(" " + text + " ", x0,
+      y0 + g.getFontMetrics().getHeight() - g.getFontMetrics().getDescent());
```
Comment (BrentEaston) [inline vassal-app/src/main/java/VASSAL/tools/image/LabelUtils.java:78]: Should this have g.addRenderingHints(SwingUtils.FONT_HINTS); to ensure the Rendering hints are in a known state?

**Review by BrentEaston — COMMENTED** [review]
I wanted to try this out with my module to see what it looks like, but it is crashing for me when trying to Play or Edit the module. This is standard Labeler usage, not Labeler.drawLabel().

java.lang.NullPointerException: null
	at VASSAL.counters.Labeler.getLabelPosition(Labeler.java:344)
	at VASSAL.counters.Labeler.boundingBox(Labeler.java:557)
	at VASSAL.counters.ReportState.boundingBox(ReportState.java:88)
	at VASSAL.counters.Marker.boundingBox(Marker.java:87)
	at VASSAL.counters.UsePrototype.boundingBox(UsePrototype.java:200)
	at VASSAL.counters.Clone.boundingBox(Clone.java:125)
	at VASSAL.counters.FreeRotator.boundingBox(FreeRotator.java:162)
	at VASSAL.build.widget.PieceSlot.getPreferredSize(PieceSlot.java:270)
	at VASSAL.build.widget.PieceSlot.getBuildElement(PieceSlot.java:467)
	at VASSAL.build.AbstractBuildable.getBuildElement(AbstractBuildable.java:184)
	at VASSAL.build.Widget.getBuildElement(Widget.java:90)
	at VASSAL.build.AbstractBuildable.getBuildElement(AbstractBuildable.java:184)
	at VASSAL.build.Widget.getBuildElement(Widget.java:90)
	at VASSAL.build.AbstractBuildable.getBuildElement(AbstractBuildable.java:184)
	at VASSAL.build.Widget.getBuildElement(Widget.java:90)
	at VASSAL.build.AbstractBuildable.getBuildElement(AbstractBuildable.java:184)
	at VASSAL.build.Widget.getBuildElement(Widget.java:90)
	at VASSAL.build.AbstractBuildable.getBuildElement(AbstractBuildable.java:184)
	at VASSAL.build.GameModule.buildString(GameModule.java:1107)
	at VASSAL.build.GameModule.updateLastSave(GameModule.java:970)
	at VASSAL.launch.EditModuleAction.loadModule(EditModuleAction.java:56)
	at VASSAL.launch.Editor.launch(Editor.java:76)
	at VASSAL.launch.Launcher$1.run(Launcher.java:150)
	at java.desktop/java.awt.event.InvocationEvent.dispatch$$$capture(InvocationEvent.java:316)

File: vassal-app/src/main/java/VASSAL/tools/image/LabelUtils.java (hunk 2 above)
Comment (uckelman) [reply vassal-app/src/main/java/VASSAL/tools/image/LabelUtils.java]: Done.

**Comment by uckelman** [comment]
I fixed the NPE and added a workaround for 8pt labels on Windows.

**Pull request merged**